src/Core/mcts/MCTSBase.cpp

```diff
@@ -9,6 +9,6 @@
 State<boost::any> state_hack(1);
 
 MCTSBase::MCTSBase(EnvironmentBase<boost::any> &environment, boost::function<int(State<boost::any>)> default_policy)
-    : env(environment), root((SearchNode*)NULL, state_hack){};
+    : env(environment), root((SearchNode*)NULL, state_hack, false){};
```

Comment:
Is it possible to remove the hack, or should we just defer this for another PR?

Reply:
Another PR. I actually made another version of MCTSBase in another branch which I think we will be using instead, where this problem doesn't arise at all